calico_node/startup/startup.go

```diff
@@ -857,6 +871,18 @@ func ensureFilesystemAsExpected() {
 		}
 	}
 
+	// Make sure the /var/lib/calico directory exists.
+	libDir := "/var/lib/calico"
+	// Check if directory already exists
+	if _, err := os.Stat(libDir); err != nil {
+		// Create the libDir
+		if err = os.MkdirAll(libDir, os.ModeDir); err != nil {
+			log.Errorf("Unable to create '%s'", libDir)
+			terminate()
+		}
+		log.Warnf("%s was not mounted, node name may not be detected properly", libDir)
```

Comment:
Something like this might give the user a bit more a of a clue of what they need to fix: `Expected %s to be mounted into the container but it wasn't present. Node name may not be detected correctly.`